tests/st/utils/utils.py

```diff
@@ -209,6 +221,10 @@ def calicoctl(command, data=None, load_as_stdin=False, format="yaml", only_stdou
         etcd_auth = "%s:2379" % ETCD_HOSTNAME_SSL
     else:
         etcd_auth = "%s:2379" % get_ip()
+
+    if broken_store :
+        etcd_auth = "broken.xyz:54321"
```

Comment:
The scenario that discovered this problem was with no etcd config at all, which might be a bit different from having a broken config - and which might also solve your problem with the failure taking a long time.  Could you adjust so that the new testing is with no etcd config at all, instead of a broken config?

Reply:
will look into that, thanks

Reply:
yup, done, added a whole bunch of test for other commands too